src/ai/defaultai.cc

```diff
@@ -1505,14 +1512,18 @@ void DefaultAI::update_all_not_buildable_fields() {
 		if (player_->get_buildcaps(unusable_fields.front()) & Widelands::BUILDCAPS_SIZEMASK) {
 			buildable_fields.push_back(new BuildableField(unusable_fields.front()));
 			unusable_fields.pop_front();
-			update_buildable_field(*buildable_fields.back());
+			if (fields_update_limit > checked_fields++) {
+				update_buildable_field(*buildable_fields.back());
```

Comment:
drive by fix: we shpuld update the field_info_expiration time as well here, to avoid a recheck to early.

same below for the mineable fields